src/datasets/summarizations/trend-partial.summarization.ts

```diff
@@ -0,0 +1,103 @@
+import * as math from 'mathjs';
+import { Summary } from './types';
+import { TimeSeriesPoint } from '../metas/types';
+import { cacheSummaries } from './utils/commons';
+import {
+  PointMembershipFunction,
+  MembershipFunction,
+  trapmf,
+  trapmfL,
+  trapmfR,
+} from './libs/protoform';
+import {
+  exponentialMovingAverage,
+  normalizedUniformPartiallyLinearEpsApprox,
+  TimeSeriesPartialTrend,
+} from './libs/trend';
+import { formatX, formatY } from '../../utils/formatters';
+
+export function queryFactory(points: TimeSeriesPoint[]) {
+  return cacheSummaries(() => {
+
+    const smoothedPoints = exponentialMovingAverage(points);
+    const partialTrends = normalizedUniformPartiallyLinearEpsApprox(smoothedPoints, 0.01);
+
+    const applyTrendAngleWithWeight = (f: MembershipFunction) => ({ cone }: TimeSeriesPartialTrend) => {
+      const avgAngleRad = (cone.endAngleRad + cone.startAngleRad) / 2;
+      return f(avgAngleRad);
+    };
+
+    const MAX_ANGLE = Math.atan(500 / 800);
+
+    const uIncreasingDynamic = applyTrendAngleWithWeight(trapmfL(MAX_ANGLE / 8, MAX_ANGLE / 4));
+    const uConstantDynamic = applyTrendAngleWithWeight(trapmf(-MAX_ANGLE / 4, -MAX_ANGLE / 8, MAX_ANGLE / 8, MAX_ANGLE / 4));
+    const uDecreasingDynamic = applyTrendAngleWithWeight(trapmfR(-MAX_ANGLE / 4, -MAX_ANGLE / 8));
+
+    const uDynamics: [string, PointMembershipFunction<TimeSeriesPartialTrend>][] = [
+      ['increased', uIncreasingDynamic],
+      ['similar', uConstantDynamic],
+      ['decreased', uDecreasingDynamic],
+    ];
+
+    const validityThreshold = 0.7;
+
+    const mergePartialTrends = (a: TimeSeriesPartialTrend, b: TimeSeriesPartialTrend): TimeSeriesPartialTrend[] => {
+      for (const [_, uDynamic] of uDynamics) {

```

Comment:
Just use a array of the functions instead of having an unused value.

for (const foo of [a, b, c]) {}